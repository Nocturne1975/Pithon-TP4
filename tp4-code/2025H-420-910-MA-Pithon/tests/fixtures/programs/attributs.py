class Counter:
    def __init__(self, start=0):
        self.count = start
        self.name = "Counter"
    
    def increment(self):
        self.count = self.count + 1
        return self.count
    
    def reset(self):
        self.count = 0
        return "Reset to " + str(self.count)

# Créer une instance
c = Counter(5)

# Accéder aux attributs
print("Initial count:", c.count)
print("Counter name:", c.name)

# Modifier un attribut directement
c.name = "MyCounter"


# Appeler des méthodes qui modifient des attributs
print("After increment:", c.increment())
print("After increment again:", c.increment())

# Ajouter un nouvel attribut
c.max_value = 100
print("Max value:", c.max_value)

# Réinitialiser
print(c.reset())
print("Final count:", c.count)
