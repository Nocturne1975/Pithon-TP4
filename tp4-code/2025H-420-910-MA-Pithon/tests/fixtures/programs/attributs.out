Initial count: 5
Counter name: Counter
After increment: 6
After increment again: 7
Max value: 100
Reset to 0
Final count: 0
